zap/src/main/java/org/zaproxy/zap/eventBus/SimpleEventBus.java

```diff
@@ -60,34 +63,33 @@ public void registerPublisher(EventPublisher publisher, String... eventTypes) {
         }
         regMgmtLock.lock();
         try {
-            if (this.nameToPublisher.get(publisher.getPublisherName()) != null) {
+            final String publisherName = publisher.getPublisherName();
+            if (nameToPublisher.get(publisherName) != null) {
                 throw new InvalidParameterException(
                         "Publisher with name "
-                                + publisher.getPublisherName()
+                                + publisherName
                                 + " already registered by "
-                                + this.nameToPublisher
-                                        .get(publisher.getPublisherName())
+                                + nameToPublisher
+                                        .get(publisherName)
                                         .getPublisher()
                                         .getClass()
                                         .getCanonicalName());
             }
-            log.debug("registerPublisher " + publisher.getPublisherName());
+            log.debug("registerPublisher " + publisherName);
 
-            RegisteredPublisher regProd = new RegisteredPublisher(publisher, eventTypes);
+            RegisteredPublisher regProd =
+                    new RegisteredPublisher(publisher, new HashSet<>(asList(eventTypes)));
 
-            List<RegisteredConsumer> consumers = new ArrayList<>();
```

Comment:
This list only exists so that we can add a bunch of consumers in one go, but we don't need this batch functionality and if we add them one by one we can remove `RegisteredPublisher.addConsumers` and simplify `RegisteredPublisher` to only have one way to add consumers.

Reply:
> but we don't need this batch functionality

It avoids changing `consumers` multiple times.

Reply:
Ah, I see - since it's a `CopyOnWriteArrayList` that's more expensive

Reply:
Any reason to not restore the previous behaviour?